ipwb/util.py

```diff
@@ -63,20 +56,18 @@ def isDaemonAlive(daemonMultiaddr=IPFSAPI_MUTLIADDRESS):
         # ConnectionError/AttributeError if IPFS daemon not running
         client.id()
         return True
-    except (ConnectionError):  # exceptions.AttributeError):
-        logError("Daemon is not running at " + daemonMultiaddr)
+
+    except ConnectionError:  # exceptions.AttributeError):
+        logger.error('Daemon is not running at %s', daemonMultiaddr)
         return False
-    except OSError:
-        logError("IPFS is likely not installed. "
-                 "See https://ipfs.io/docs/install/")
-        sys.exit()
-    except Exception as e:
-        logError('Unknown error in retrieving daemon status')
-        logError(sys.exc_info()[0])
 
+    except OSError:
+        raise Exception(
+            'IPFS is likely not installed. See https://ipfs.io/docs/install/'
+        )
 
-def logError(errIn):
-    print(errIn, file=sys.stderr)
+    except Exception as e:
+        raise Exception('Unknown error in retrieving daemon status.') from e
```

Comment:
Will the removal of `logError(sys.exc_info()[0])` provide a sufficient traceback for debugging?

Reply:
Here is an example:
```pycon
In [1]: try: 
   ...:     raise ValueError('Ugly internal error, GRRRRR!!!') 
   ...: except ValueError as err: 
   ...:     raise Exception('Nice user-friendly error :)') from err 
   ...:                                                                                                           
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-1-8ed378f4cf49> in <module>
      1 try:
----> 2     raise ValueError('Ugly internal error, GRRRRR!!!')
      3 except ValueError as err:

ValueError: Ugly internal error, GRRRRR!!!

The above exception was the direct cause of the following exception:

Exception                                 Traceback (most recent call last)
<ipython-input-1-8ed378f4cf49> in <module>
      2     raise ValueError('Ugly internal error, GRRRRR!!!')
      3 except ValueError as err:
----> 4     raise Exception('Nice user-friendly error :)') from err
      5 

Exception: Nice user-friendly error :)
```

As you can see, the user will see the error message you supplied in `raise`, together with its stack trace. And if the user is to scroll a bit upper they will see the original exception properly preserved.

Reply:
Ok, I am convinced. PEP 415's`raise exc from None` will also help suppress context, which is useful for development but noisy for the end user who should just be told directly what is wrong. I am fine using the pattern provided in this PR.